include/neighbor.h

```diff
@@ -78,7 +78,7 @@ class NeighborPriorityQueue
             }
         }
 
-        if (lo < _capacity)
+        if (lo < _size)
```

Comment:
@SMovaghati to confirm my understanding, this change avoids calling `memmove(..., ..., 0)` which is a logical no-op when 1) the new element is appended to the end of the queue and 2) the queue is not full. Is that right?

(Only in this scenario `_lo == _size`, otherwise always `lo < _size`)

Reply:
That is correct. If we are not adding an item at the end of the queue, then we always need the memmove to open up a space in the middle.